Source/HBIOS/ps2iface.inc

```diff
@@ -0,0 +1,247 @@
+;__________________________________________________________________________________________________
+; HARDWARE INTERFACE FOR POLLED AND INTERRUPT DRIVEN PS2 KEYBOARD.
+;
+; IN POLLED MODE THE KEYBOARD DEVICE IS IDENTIFIED BY THE TABLE POINTED TO BY IY
+; IN INTERRUPT MODE THE KEYBOARD DEVICE IS UNKNOWN SO PREDEFINED PORT VALUES ARE USED.
+;
+; REQUIRES AN APPROPRIATE INTERRUPT JUMPER TO BE SET. 
+; CURRENTLY CONFIGURED FOR INT #2 ON DUODYNE WITH I/O MULTI BOARD.
+;__________________________________________________________________________________________________
+;
+#IF ((INTMODE == 2) & INTPS2KBD))
+
+#IF (TMSENABLE)
+KBDPORTD	.DB	TMS_KBDDATA	; KBD CTLR DATA PORT
+KBDPORTS	.DB	TMS_KBDST	; KBD CTLR STATUS/CMD PORT	
+#ELSE
+
+#IF (GDCENABLE)
+KBDPORTD	.DB	GDC_KBDDATA	; KBD CTLR DATA PORT
+KBDPORTS	.DB	GDC_KBDST	; KBD CTLR STATUS/CMD PORT
+#ELSE
+
+#IF (CVDUENABLE)
+KBDPORTD	.DB	CVDU_KBDDATA	; KBD CTLR DATA PORT
+KBDPORTS	.DB	CVDU_KBDST	; KBD CTLR STATUS/CMD PORT
+#ELSE
+
+#IF (VRCENABLE)
+KBDPORTD	.DB	VRC_KBDDATA	; KBD CTLR DATA PORT
+KBDPORTS	.DB	VRC_KBDST	; KBD CTLR STATUS/CMD PORT
+#ELSE
+
+#IF (VGAENABLE)
+KBDPORTD	.DB	VGA_KBDDATA	; KBD CTLR DATA PORT
+KBDPORTS	.DB	VGA_KBDST	; KBD CTLR STATUS/CMD PORT
+#ENDIF
+
+#ENDIF
+#ENDIF
+#ENDIF
+#ENDIF
+;
+;__________________________________________________________________________________________________
+; KEYBOARD QUEUE FOR INTERRUPTS
+;__________________________________________________________________________________________________
+;
+;
+KBDQSIZ		.EQU	16		; MUST BE POWER OF TWO, <=128
+KBDQGETX	.DB	0		; QUEUE INDEX
+KBDQPUTX	.DB	0		; QUEUE INDEX
+		.DB	$55		; GUARD BYTE FOR DEBUGGING
+KBDQ		.FILL	KBDQSIZ,0	; CIRCULAR KEYBOARD BUFFER
+		.DB	$AA		; GUARD BYTE FOR DEBUGGING
+
+;__________________________________________________________________________________________________
+; RESET THE QUEUE TO EMPTY
+;__________________________________________________________________________________________________
+;
+KBDQINIT: 
+	XOR	A
+	LD	(KBDQGETX),A
+	LD	(KBDQPUTX),A
+	RET
+;
+;__________________________________________________________________________________________________
+; KEYBOARD INTERRUPT VECTOR
+;__________________________________________________________________________________________________
+;
+KBD_INT:
+;	CALL	KBDQDBG
+	LD	A,(KBDPORTS)		; READ CONTROLLER STATUS
+	LD	C,A
+	IN	A,(C)

```

Comment:
It looks like the EZ80_IO macro is missing from the new I/O calls.

Reply:
Fixed. Thanks.